src/xarray_regrid/utils.py

```diff
@@ -238,6 +256,11 @@ def format_for_regrid(
         "lat": {"names": ["lat", "latitude"], "func": format_lat},
         "lon": {"names": ["lon", "longitude"], "func": format_lon},
     }
+
+    if stats:
+        coord_handlers.pop("lat")  # padding lat with stats methods is problematic.
```

Comment:
@slevang I added this to disable the polar padding (should not be applied to the zonal statistic data at all).

I did not want to disable the longitude padding, however, it causes casting integer data types to float64 dtypes. I don't have time to solve this right now though.